modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -287,18 +288,30 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
           .asScala
           .zipWithIndex
           .map { case ((name, value), index) =>
-            EnumValue(value, index, name)
+            val member = shape.getMember(name).get()
+
+            // random note: any other traits that are allowed on enum values? cuz they don't generate hints.
+            // known: @deprecated, @documentation
+            EnumValue(value, index, name, hints(member))
           }
           .toList
-        Enumeration(shape.getId(), shape.name, values).some
+
+        Enumeration(
+          shape.getId(),
+          shape.name,
+          values,
+          hints = hints(shape)
+        ).some
       }
 
       override def intEnumShape(shape: IntEnumShape): Option[Decl] = {
         val values = shape
           .getEnumValues()
           .asScala
           .map { case (name, value) =>
-            EnumValue(name, value, name)
+            val member = shape.getMember(name).get()
```

Comment:
feels a bit odd to do the lookup like this, but it's not _that_ much repetition and it _should_ be safe at all times.